### DISCUSSION TITLE
Is there any way to not publish any ports when setting up AIO?

### DISCUSSION THREAD

**Discussion by max-critcrew** [nextcloud/all-in-one#4238] [Ideas] [answered]
Hey everyone!

We're trying to set up nextcloud-AIO behind a VPN and we do not want any ports to be published at all. From our testing when the Apache server binding a loopback port leads to the port being open to the outside. 

Tested via:
`sudo nmap -p- <external_server_ip>
`

Our Apache server does not need to connect to the outside, since we've setup traefik with dns challenge in a way that seems to work just fine. If we just had a flag/setting to never publish any ports at all (neither with Apache nor with Talk), that'd be amazing. Any idea what we could do to achieve that?

**Comment by szaimen** [comment]
Hi, see https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#3-limit-the-access-to-the-apache-container

**Comment by max-critcrew** [comment]
@szaimen Thank you, I did a full new install (carefully went through the reset instructions! :) ) and set it to 127.0.0.1 but it still seems to publish the 11000 port to the outside. Additionally the talk container also still publishes its ports.
Any more ideas why it still publishes ports and/or how to make Talk stop publishing ports?

**Comment by szaimen** [comment, marked as answer]
Sorry, apparently I wasnt clear enough. 

> Is there any way to not publish any ports when setting up AIO?

No, but you can limit the apache container to localhost as I mentioned in https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#3-limit-the-access-to-the-apache-container. The Talk port cannot be limited.

Also even though this sounds like a nice feature, the requests for this are quite low. Currently there are no plans to implement such a feature. 

If you need such control, you might want to have a look at https://github.com/nextcloud/all-in-one/tree/main/manual-install.

**Reply by max-critcrew** [reply]
Thank you, understood now! Also completely understandable that it's not worth it to implement this if there's no big need for it.

We're running Nextcloud behind a VPN + traefik, so for us there's no reason to open ports to the outside world - but we also didn't want to lose the AIO backups and easy to setup-goodness :)
